date.go

```diff
@@ -174,15 +174,16 @@ func (d *dateObject) timeUTC() time.Time {
 	return timeFromMsec(d.msec).In(time.UTC)
 }
 
-func (d *dateObject) MemUsage(ctx *MemUsageContext) (uint64, error) {
+func (d *dateObject) MemUsage(ctx *MemUsageContext) (memUsage uint64, newMemUsage uint64, err error) {
 	if d == nil || ctx.IsObjVisited(d) {
-		return SizeEmpty, nil
+		return SizeEmpty, SizeEmpty, nil
 	}
 	ctx.VisitObj(d)
 
 	// start with the size of msec
-	total := SizeNumber
-	inc, err := d.baseObject.MemUsage(ctx)
-	total += inc
-	return total, err
+	memUsage = SizeNumber
+	newMemUsage = SizeNumber
+	inc, newInc, err := d.baseObject.MemUsage(ctx)
+
+	return memUsage + inc, newMemUsage + newInc, err
```

Comment:
[nit] Can we add a test to check these return changes?

Reply:
Do you mean a case with a non-empty baseObject (which I noticed was missing)?

Reply:
perfect, will add a test!